src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -117,14 +120,14 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord)
 
                 if (!string.IsNullOrEmpty(logRecord.EventId.Name))
                 {
-                    otlpLogRecord.Attributes.AddStringAttribute(nameof(logRecord.EventId.Name), logRecord.EventId.Name);
+                    otlpLogRecord.Attributes.AddStringAttribute(nameof(logRecord.EventId.Name), logRecord.EventId.Name, attributeValueLengthLimit);
                 }
 
                 if (logRecord.Exception != null)
                 {
-                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionType, logRecord.Exception.GetType().Name);
-                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionMessage, logRecord.Exception.Message);
-                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionStacktrace, logRecord.Exception.ToInvariantString());
+                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionType, logRecord.Exception.GetType().Name, attributeValueLengthLimit);
+                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionMessage, logRecord.Exception.Message, attributeValueLengthLimit);
+                    otlpLogRecord.Attributes.AddStringAttribute(SemanticConventions.AttributeExceptionStacktrace, logRecord.Exception.ToInvariantString(), attributeValueLengthLimit);
```

Comment:
I suggest we get rid of the `AddStringAttribute` method and use `OtlpKeyValueTransformer.Instance.TryTransformTag` even for these one-offs. That way we benefit from the uniform way we process string attributes.

Also, OTLP tag transform has a test suite, so the truncation logic is at least covered there. It'd be nice to also have some tests specifically for the log exporter logic. That said, we do not yet have tests for the trace exporter either with respect to attribute limits.

Reply:
I didn't removed the AddStringAttribute function, but replaced the implementation to use the OtlpKeyValueTransformer.Instance.TryTransformTag.
Removing the function itself was for me to excessive (additional if for the TryTransform and for the AttributeCount). 
Is that good for you?

About tests: I will check during the next days when I have a free timeslot. 👍